src/structs.h

```diff
@@ -104,7 +104,7 @@ struct Node
                (Parent == other.Parent) &&
                (interval == other.interval);
     }
-    auto hash_value(Node const& n) -> std::size_t{
```

Comment:
this should be a const function, I guess, otherwise it wouldn't match with the unordered_set hashvalue function signature?

Reply:
@dwthomas can you change this to a const function and do a quick pull request? I have no idea why it can compile on your side. have you tried to relax some compile flags in your local branch? We should treat all warnings as errors.